torch/utils/cpp_extension.py

```diff
@@ -145,6 +145,8 @@ def _find_sycl_home() -> Optional[str]:
     """Find the OneAPI install path."""
     # Guess #1
     sycl_home = os.environ.get('ONEAPI_ROOT')
+    if sycl_home is not None:
+        sycl_home = os.path.join(sycl_home, "compiler", "latest")
```

Comment:
this is true for DLE package

Reply:
@etaf which package is supposed to be installed? So that `icpx` is available at the path: `/opt/intel/oneapi/bin/icpx`. Could you take a look?

According to https://github.com/intel/intel-xpu-backend-for-triton/pull/2952#issuecomment-2523478961 possible locations:
* `/opt/intel/oneapi/compiler/2024.1/bin/icpx`, `/opt/intel/oneapi/pytorch-gpu-dev-0.5/bin/icpx` in PTDB
* `/opt/intel/oneapi/2025.0/bin/icpx`, `/opt/intel/oneapi/compiler/2025.0/bin/icpx` in DLE

Reply:
For Pytorch 2.6, ONEAPI_ROOT=/opt/intel/oneapi/2025.0, so without this patch, it can should able to find `SYCL_HOME`/bin/icpx.

And I'm curious that why you need to find the icpx compiler, this is only used by Inductor Cpp Wrapper now. Do you know why?

Further more, we're working on replaceing icpx with g++ for Inductor Cpp Wrapper to exclude the dependency on OneAPI Compiler in runtime.

Reply:
> For Pytorch 2.6, ONEAPI_ROOT=/opt/intel/oneapi/2025.0

Source for that statement?

After installing Deep Learning Essentials with the offline installer (https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html?packages=dl-essentials&dl-essentials-os=linux&dl-lin=offline):

```
$ source /opt/intel/oneapi/setvars.sh
...
$ echo $ONEAPI_ROOT
/opt/intel/oneapi
```

Reply:
You may source the wrong env var:
```
source /opt/intel/oneapi/2025.0/oneapi-vars.sh
echo $ONEAPI_ROOT
/opt/intel/oneapi/2025.0
```

And to be more common, you can also source like this (https://github.com/pytorch/pytorch/blob/b1bb860d3c812371b89a9725407230216e7369b5/.ci/manywheel/build_xpu.sh#L19-L22):
```
source /opt/intel/oneapi/compiler/latest/env/vars.sh
 echo $ONEAPI_ROOT
(ONEAPI_ROOT is not set)
which icpx
/opt/intel/oneapi/compiler/2025.0/bin/icpx
```
with the path of icpx, we can also find the sycl home.

Reply:
> `source /opt/intel/oneapi/2025.0/oneapi-vars.sh`

Will try that, thanks. Initializing with `/opt/intel/oneapi/setvars.sh` worked for us until recently, so this is a regression.

Reply:
I've refined the find_sycl_home in PR #142322  to be compatible with this situation.